pkg/controller/nstemplateset/nstemplatetier.go

```diff
@@ -56,26 +46,30 @@ func getTemplatesFromHost(hostClusterFunc cluster.GetHostClusterFunc, tierName s
 		return nil, fmt.Errorf("the host cluster is not ready")
 	}
 
-	tier := &toolchainv1alpha1.NSTemplateTier{}
+	tierTemplate := &toolchainv1alpha1.TierTemplate{}
 	err := host.Client.Get(context.TODO(), types.NamespacedName{
 		Namespace: host.OperatorNamespace,
-		Name:      tierName,
-	}, tier)
+		Name:      templateRef,
+	}, tierTemplate)
 	if err != nil {
-		return nil, errors.Wrapf(err, "unable to retrieve the NSTemplateTier '%s' from 'Host' cluster", tierName)
-	}
-	result := templates{}
-	for _, ns := range tier.Spec.Namespaces {
-		result[ns.Type] = versionedTemplate{
-			Revision: ns.Revision,
-			Template: ns.Template,
-		}
+		return nil, errors.Wrapf(err, "unable to retrieve the TierTemplate '%s' from 'Host' cluster", templateRef)
 	}
-	if tier.Spec.ClusterResources != nil {
-		result[ClusterResources] = versionedTemplate{
-			Revision: tier.Spec.ClusterResources.Revision,
-			Template: tier.Spec.ClusterResources.Template,
-		}
-	}
-	return result, nil
+	return tierTemplate, nil
+}
+
+// tierTemplate contains all data from TierTemplate including its name
+type tierTemplate struct {
+	templateRef string
+	tierName    string
+	typeName    string
+	revision    string
+	template    templatev1.Template
```

Comment:
It looks like you don't need the `revision` field (other fields are used, AFAICS).

Reply:
yeah, I copied all values from the `TierTemplate`

Reply:
removed 758cca5